### PR TITLE
Document logging via Monolog

### PR DESCRIPTION
**Pull request by SeverinGloeckle**
Add documentation for working with Monolog in Contao, the use of `ContaoContext` and the new convenience logger services from https://github.com/contao/contao/pull/3702.

### CODE REVIEW

File: docs/dev/framework/logging.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+Monolog provides a standardized and simple, yet flexible and extensible solution
+to log messages and process them through a variety of sources.
+
+You can use all features of Monolog and the Monolog Bundle in your Contao application.
+
+## Log Channels
+In more complex applications, log channels are used to group related messages and
+optionally process them differently.
+
+A typical Symfony application typically uses channels such as `request` or `security`
```
Comment (Toflar) [inline docs/dev/framework/logging.md]:
```suggestion
A Symfony application typically uses channels such as `request` or `security`
```

File: docs/dev/framework/logging.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+```php
+use Contao\CoreBundle\Monolog\ContaoContext;
+
+$logger->info(
+    'This message is intended for the Contao system log', 
+    ['contao' => new ContaoContext(__METHOD__, ContaoContext::GENERAL)]
+);
+```
+
+All messages with this `ContaoContext` are automatically send to the system log
```
Comment (Toflar) [inline docs/dev/framework/logging.md]:
```suggestion
All messages with this `ContaoContext` are automatically sent to the system log
```

**Review by Toflar — APPROVED** [review]
Minor nitpicks but a great addition <3

File: docs/dev/framework/logging.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+Contao itself uses the `contao` channel for all Contao-specific messages.
+
+{{% notice note %}}
+Contao also creates additional log channels for sending messages with a specific context.
+For more information, see [Contao channels](#contao-channels-convenience-loggers).
+{{% /notice %}}
+
+## Log Contexts & Contao's System Log
+
+The system log in the Contao backend lists all log entries that have been specifically flagged for Contao.
```
Comment (fritzmg) [inline docs/dev/framework/logging.md]:
```suggestion
The system log in the Contao back end lists all log entries that have been specifically flagged for Contao.
```

**Review by fritzmg — CHANGES_REQUESTED** [review]

**Review by fritzmg — APPROVED** [review]

**Comment by fritzmg** [comment]
Thank you very much @SeverinGloeckle :)

**Pull request merged**